std/json.zig

```diff
@@ -0,0 +1,1308 @@
+// JSON parser conforming to RFC8259.
+//
+// https://tools.ietf.org/html/rfc8259
+
+const std = @import("index.zig");
+const mem = std.mem;
+
+const u1 = @IntType(false, 1);
+const u256 = @IntType(false, 256);
+
+pub const TokenId = enum(u5) {
+    ObjectBegin,
+    ObjectEnd,
+    ArrayBegin,
+    ArrayEnd,
+    String,
+    Number,
+    True,
+    False,
+    Null,
+};
+
+// A single token slice into the parent string.
+//
+// Use `token.slice()` on the inptu at the current position to get the current slice.
+pub const Token = struct {
+    id: TokenId,
```

Comment:
it would be idiomatic to rename `TokenId` to `Id` and nest it inside `Token`. Then it can be referred to with `Token.Id`